src/dd/StateGeneration.cpp

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2023 - 2025 Chair for Design Automation, TUM
+ * Copyright (c) 2025 Munich Quantum Software Company GmbH
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * Licensed under the MIT License
+ */
+
+#include "dd/StateGeneration.hpp"
+
+#include "dd/Complex.hpp"
+#include "dd/ComplexNumbers.hpp"
+#include "dd/DDDefinitions.hpp"
+#include "dd/Edge.hpp"
+#include "dd/Node.hpp"
+#include "dd/Package.hpp"
+#include "ir/Definitions.hpp"
+
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <cmath>
+#include <complex>
+#include <cstddef>
+#include <iterator>
+#include <numeric>
+#include <random>
+#include <stdexcept>
+#include <utility>
+#include <vector>
+
+namespace dd {
+namespace {
+using Generator = std::mt19937_64;
+using AngleDistribution = std::uniform_real_distribution<double>;
+using IndexDistribution = std::uniform_int_distribution<std::size_t>;
+
+/**
+ * @brief Return random complex number on the unit circle, such that,
+ * its squared magnitude is one.
+ */
+std::complex<double> randomComplexOnUnitCircle(Generator& gen,
+                                               AngleDistribution& dist) {
+  const double angle = dist(gen);
+  return {std::cos(angle), std::sin(angle)};
+}
+
+/**
+ * @brief Generate node with edges pointing at @p left and @p right.
+ * Initialize edge weights randomly with the constraint that their norm is one.
+ * @note The CachedEdge ensures that the weight of the resulting edge is not
+ * stored in the lookup table.
+ */
+vCachedEdge randomNode(Qubit v, vNode* left, vNode* right, Generator& gen,
+                       AngleDistribution& dist, Package& dd) {
+  const auto alpha = randomComplexOnUnitCircle(gen, dist) * SQRT2_2;
+  const auto beta = randomComplexOnUnitCircle(gen, dist) * SQRT2_2;
+
+  const std::array<vCachedEdge, RADIX> edges{
+      vCachedEdge(left, ComplexValue(alpha)),
+      vCachedEdge(right, ComplexValue(beta))};
```

Comment:
```suggestion
ComplexValue randomComplexOnUnitCircle(Generator& gen,
                                               AngleDistribution& dist) {
  const double angle = dist(gen);
  return {std::cos(angle), std::sin(angle)};
}

/**
 * @brief Generate node with edges pointing at @p left and @p right.
 * Initialize edge weights randomly with the constraint that their norm is one.
 * @note The CachedEdge ensures that the weight of the resulting edge is not
 * stored in the lookup table.
 */
vCachedEdge randomNode(Qubit v, vNode* left, vNode* right, Generator& gen,
                       AngleDistribution& dist, Package& dd) {
  const auto alpha = randomComplexOnUnitCircle(gen, dist) * SQRT2_2;
  const auto beta = randomComplexOnUnitCircle(gen, dist) * SQRT2_2;

  const std::array<vCachedEdge, RADIX> edges{
      vCachedEdge(left, alpha),
      vCachedEdge(right, beta)};
```
Just slightly more ergonomic.